src/Checkers/LicenseChecker.js

```diff
@@ -0,0 +1,303 @@
+// MIT License
+//
+// Copyright 2017 Electric Imp
+//
+// SPDX-License-Identifier: MIT
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const minimatch = require('minimatch');
+const colors = require('colors/safe');
+
+const checker = require('./AbstractChecker');
+
+const LICENSE_FILE_PATH = './resources/LICENSE.example';
+
+const TOKENS = {
+  END: 'END_OF_LICENSE',
+};
+
+class ErrorMessage {
+
+  constructor(message, lineNum, file, checker, linePos = -1) {
+    this._line = lineNum;
+    this._linePos = linePos;
+    this._message = message;
+    this._file = file;
+    this._checker = checker;
+  }
+
+  toString() {
+    return colors.red(`${this._checker} Error:`) + '\n' +
+            `\t${this._message}` + '\n' +
+            colors.red('\tin ') +  this._file + '\n' +
+            colors.red('\tat ') + `${this._line}` + (this._linePos === -1 ? '' : `:${this._linePos}`);
+  }
+
+  toShortString() {
+    return colors.blue(`${this._checker} Error in `) + this._file + colors.blue(` at ${this._line}` + (this._linePos === -1 ? '' : `:${this._linePos}`));
+  }
+}
+
+class LicenseChecker extends checker {
+
+  constructor(exclude = null) {
+    super();
+    this.excludeList = exclude;
+    this._extensionsSet = new Set(['.js', '.nut']);
+  }
+
+
+  /**
+   * Retrieves the next token (word from the file)
```

Comment:
there is no concept of file for this function. is accepts a string and a flag.
`Retrieves a next token (word) from the specified text`